client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts

```diff
@@ -55,6 +55,22 @@ const debug = debugFactory( 'calypso:composite-checkout:get-thank-you-page-url'
 type SaveUrlToCookie = ( url: string ) => void;
 type GetUrlFromCookie = () => string | undefined;
 
+type OrderId = number;
+type PurchaseId = number;
+type ReceiptId = number;
+type ReceiptIdPlaceholder = ':receiptId';
+type PendingOrderPathFragment = `pending/${ OrderId }`;
+type PendingOrReceiptId = PendingOrderPathFragment | ReceiptIdPlaceholder | PurchaseId | ReceiptId;
+
+function isReceiptIdOrPlaceholder(
+	receiptId: PendingOrReceiptId
+): receiptId is ReceiptId | ReceiptIdPlaceholder {
+	if ( Number.isInteger( receiptId ) || receiptId === ':receiptId' ) {
+		return true;
+	}
+	return false;
+}
+
 /**
```

Comment:
I think there is a bug here. The function is returning false even when receiptId is a number. We believe this is preventing users from activating their products at the end of the checkout flow. See p1657036519941789-slack-CQXNTE9K9.

Reply:
Thanks! Fixed by https://github.com/Automattic/wp-calypso/pull/65271